extensions/amp-image-slider/0.0/amp-image-slider.js

```diff
@@ -0,0 +1,566 @@
+    this.rightLabelWrapper_ = null;
+    /** @private {?Element} */
+    this.rightLabel_ = null;
+
+    /** @private {!Element} */
+    this.leftMask_ = this.doc_.createElement('div');
+
+    /** @private {!Element} */
+    this.rightMask_ = this.doc_.createElement('div');
+
+    /** @private {!Element} */
+    this.bar_ = this.doc_.createElement('div');
+
+    /** @private {!Element} */
+    this.barStick_ = this.doc_.createElement('div');
+
+    /** @private {?Element} */
+    this.barButton_ = null;
+    /** @private {?Element} */
+    this.barButtonIcon_ = null;
+    /** @private {?Element} */
+    this.barHint_ = null;
+    /** @private {boolean} */
+    this.isBarHintHidden_ = false;
+
+    /** @private {number} */
+    this.moveOffset_ = 0;
+
+    /** @private {number} */
+    this.splitOffset_ = 0;
+
+    // Bind this of handlers
+    this.handleHover = this.handleHover.bind(this);
+    this.handleClickImage = this.handleClickImage.bind(this);
+    this.handleTapImage = this.handleTapImage.bind(this);
+    this.handleHideHint = this.handleHideHint.bind(this);
+
+    this.dragStart = this.dragStart.bind(this);
+    this.dragMove = this.dragMove.bind(this);
+    this.dragEnd = this.dragEnd.bind(this);
+
+    this.touchStart = this.touchStart.bind(this);
+    this.touchMove = this.touchMove.bind(this);
+    this.touchEnd = this.touchEnd.bind(this);
+
+    this.pointerMoveX_ = this.pointerMoveX_.bind(this);
+
+    /** @private {boolean} */
+    this.isEventRegistered_ = false;
+  }
+
+  /** @override */
+  buildCallback() {
+    // From https://github.com/ampproject/amphtml/pull/16688
+    user().assert(isExperimentOn(this.win, 'amp-image-slider'),
+        'Experiment <amp-image-slider> disabled');
+
+    const children = this.getRealChildren();
+
+    for (let i = 0; i < children.length; i++) {
+      const child = children[i];
+      if (child.hasAttribute('before')) {
+        switch (child.tagName.toLowerCase()) {
+          case 'amp-img':
+            this.leftAmpImage_ = child;
+            break;
+          case 'div':
+            this.leftLabel_ = child;
+            break;
+        }
+      } else if (child.hasAttribute('after')) {
+        switch (child.tagName.toLowerCase()) {
+          case 'amp-img':
+            this.rightAmpImage_ = child;
+            break;
+          case 'div':
+            this.rightLabel_ = child;
+            break;
+        }
+      }
+    }
+
+    if (!this.leftAmpImage_ || !this.rightAmpImage_) {
+      return null;
+    }
+
+    if (!isExperimentOn(this.win, 'layers')) {
+      // When layers not enabled
+      this.setAsOwner(this.leftAmpImage_);
+      this.setAsOwner(this.rightAmpImage_);
+    }
+
+    this.container_.classList.add('i-amphtml-image-slider-container');
+    this.element.appendChild(this.container_);
+
+    this.buildImages();
+    this.buildBar();
+
+    return Promise.resolve();
+  }
+
+  /**
+   * Build image structures
+   */
+  buildImages() {
```

Comment:
Private functions should contain a trailing underscore. E.g. `buildImages_()` and marked as `@private` via type check.